rustls/src/server/server_conn.rs

```diff
@@ -185,8 +185,11 @@ impl<'a> ClientHello<'a> {
 
 /// Common configuration for a set of server sessions.
 ///
-/// Making one of these can be expensive, and should be
-/// once per process rather than once per connection.
+/// While creation of a `ServerConfig` was previously documented to be expensive, this has
+/// recently been imprecise. The largest potential cost in building a `ServerConfig` is in
```

Comment:
```suggestion
/// recently become imprecise. The largest potential cost in building a `ServerConfig` is in
```

"recently been imprecise" is valid but reads weirdly to me because "recently" usually goes with a discrete moment in time rather than a continuous interval.

Coming to think of it, though - hasn't it always been the case that gathering the roots was the only expensive part?

How about:

> Making one of these is cheap, though one of the inputs may be expensive: gathering trust roots from the operating system (to call [`with_root_certificates`]) may take on the order of a few hundred milliseconds.

(Note that this doesn't explicitly say that it used to be expensive, which I think is okay because we're replacing the expensiveness text with other text that talks about expensiveness, rather than deleting it entirely).

Reply:
Nice, I've used the language you've suggested.